### ISSUE TITLE
[no-obj-calls] Duplicates TypeScript's TS2349 error

### ISSUE THREAD

**Issue by OliverSieweke** [typescript-eslint/typescript-eslint#999]
Recommended ESLint rules that are already handled by TypeScript are turned off in `plugin:@typescript-eslint/recommended`. 
The recommended [`no-obj-calls`](https://eslint.org/docs/rules/no-obj-calls) rule should be included in the list as it triggers an ESLint as well as a TypeScript error.

**Repro**

```JSON
{
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended"
  ]
}
```

```TS
Math();
JSON();
Reflect();
Atomics();
```

**Expected Result**

A simple TypeScript error on each line:

- `TS2349: This expression is not callable. Type [...] has no call signatures.`

**Actual Result**

A duplicate error for each line, one coming from TypeScript the other one coming from ESLint: 

- `TS2349: This expression is not callable. Type [...] has no call signatures.`
- `ESLint: '[...]' is not a function.(no-obj-calls)`

**Additional Info**

Related: #983

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.0` |
| `@typescript-eslint/parser`        | `2.3.0` |
| `TypeScript`                       | `3.6.3` |
| `ESLint`                           | `6.4.0` |
| `node`                             | `10.16.3` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
consolidated eslint-recommended changes in #1273
